common/transport/amqp/test/_amqp_test.js

```diff
@@ -163,6 +163,46 @@ describe('Amqp', function () {
       });
     });
 
+    /*Tests_SRS_NODE_COMMON_AMQP_99_043: [The `connect` method shall set the `config.sslOptions.agent` option based on the `amqp.webSocketAgent` object passed in the `options` structure via the `setOptions` function.]*/
+    it('uses the agent passed into setOptions', function (testCallback) {
+      var actualAgent;
+      var fakeAgent = '__FAKE_AGENT__';
+      var amqp = new Amqp();
+      sinon.stub(amqp._rheaContainer, 'websocket_connect').callsFake(() => {
+        return sinon.stub().callsFake((uri, protocols, options) => {
+          actualAgent = options.agent;
+        });
+      });
+      var fakeConnection = new EventEmitter();
+      fakeConnection.name = 'connection';
+      var fakeConnectionContext = {connection: fakeConnection};
+      sinon.stub(amqp._rheaContainer, 'connect').callsFake(() => {
+        process.nextTick(() => {
+          amqp._rheaConnection.emit('connection_open', fakeConnectionContext);
+        });
+        return fakeConnection;
+      });
+      var fakeSession = new EventEmitter();
+      var fakeSessionContext = {session: fakeSession};
+      fakeConnection.create_session = sinon.stub().returns(fakeSession);
+      fakeSession.open = () => {};
```

Comment:
nit: is this line necessary when you're stubbing the open function in the next line? Or since you're creating this fake session could you not even stub it, just use this line to define the same behavior as the stub in the next line?

Reply:
No longer relevant.